test/clients/services_manager_test.rb

```diff
@@ -22,20 +22,76 @@
 
 require_relative '../test_helper'
 
+require "y2journal"
+
 require "yast"
 require "services-manager/clients/services_manager"
 
 describe Y2ServicesManager::Clients::ServicesManager do
   include Yast::UIShortcuts
 
-  # Finds in the widgets tree a widget with the given id
+  # Checks whether a widgets tree contains the given id
   #
   # @param tree [Yast::Term]
   # @param id [Symbol]
   #
   # @return [Boolean]
-  def exist_widget?(tree, id)
-    !tree.nested_find { |w| w.is_a?(Yast::Term) && w.value == :id && w[0] == id }.nil?
+  def contain_widget?(tree, id)
```

Comment:
These methods look like candidates to be extracted to yast2 helpers. It is out of the scope of this PR, but we should take it into account.

Reply:
Yes, they would be helpful.